From 2860168ac3513d416b8bbfee2ae2c3e279036b2a Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 13 May 2024 11:51:21 +0200
Subject: [PATCH 0886/1695] dts: stm32: add a new SCMI agent for CortexM33 on
 stm32mp25 platform

It uses IPCC channel 16, and shares only vdd33ucpd regulator.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fc3078abab201f5649f3d6e9c0ebf9697d2c242c dts: stm32: add a new SCMI agent for CortexM33 on stm32mp25 platform]
Co-developed-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I898663967d7cb23bf56c735d29a9d7a1c7afbc4c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/380390
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 23 +++++++++++++++++++----
 1 file changed, 19 insertions(+), 4 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 1d5323b58..22183563d 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -157,10 +157,6 @@
 			reg = <VOLTD_SCMI_VDDIO4>;
 			voltd-supply = <&vddio4>;
 		};
-		voltd-vdd33ucpd {
-			reg = <VOLTD_SCMI_UCPD>;
-			voltd-supply = <&vdd33ucpd>;
-		};
 		voltd-vdda18adc {
 			reg = <VOLTD_SCMI_ADC>;
 			voltd-supply = <&vdda18adc>;
@@ -171,6 +167,25 @@
 		};
 	};
 
+	scmi_cm33_agent: scmi-cm-agent@0 {
+		compatible = "st,scmi-cortexm-agent";
+		mbox-names = "scmi_cm33";
+		mboxes = <&ipcc1 RIF_IPCC_CPU1_CHANNEL(16)>;
+		scmi-agent-id = <1>;
+		scmi-channel-id = <0>;
+
+		scmi_regu_cm33: scmi-regu-cm33 {
+			compatible = "st,scmi-regulator-consumer";
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			voltd-vdd33ucpd {
+				reg = <VOLTD_SCMI_UCPD>;
+				voltd-supply = <&vdd33ucpd>;
+			};
+		};
+	};
+
 	soc@0 {
 		compatible = "simple-bus";
 		#address-cells = <1>;
-- 
2.47.2

